src/ucp/core/ucp_request.h

```diff
@@ -54,10 +54,11 @@ enum {
     UCP_REQUEST_FLAG_RKEY_INUSE            = UCS_BIT(18),
     UCP_REQUEST_FLAG_USER_HEADER_COPIED    = UCS_BIT(19),
     UCP_REQUEST_FLAG_USAGE_TRACKED         = UCS_BIT(20),
+    UCP_REQUEST_FLAG_INVALIDATED           = UCS_BIT(21),
```

Comment:
- pls move under `UCS_ENABLE_ASSERT` section
- need better name since it's irrelevant for any request, e.g. `UCP_REQUEST_FLAG_MEMH_INVALIDATED`

Reply:
In this PR it's used only within `ucs_assert`, but in the upcoming invalidation PR I use this flag in a couple of places to check whether request had been already invalidated. But I'll check once again if I manage to get rid those checks. So far it's not easy, because in some parts of the code after invalidation we still access `rndv` part.

This flag indicates invalidated request (not just memory handle), meaning the request which populated `invalidate` struct from the `send` union. As a consequence, we cannot safely access other union structures from invalidated request (e.g. we can't anymore safely access `rndv` part). So in fact this is the property of request itself

Reply:
- lets move back when and IF it's really needed
 - we dont have dedicated flags for union members, all prevoius flags named by state and/or action, IMO "invalidated request " may have to wide meaning without context

Reply:
I think I will add code to this PR that actually uses this flag in production code path

Reply:
done